tests/test.xpi.js

```diff
@@ -0,0 +1,263 @@
+    var myXpi = new Xpi('foo/bar', this.fakeZipLib);
+
+    this.openStub.yieldsAsync(null, this.fakeZipFile);
+    this.openReadStreamStub.yields(new Error('openReadStream test'));
+    this.entryStub.yields(chromeManifestEntry);
+
+    return myXpi.getMetaData()
+      .then(() => {
+        assert.fail(null, null, 'Unexpected success');
+      })
+      .catch(err => {
+        assert.include(err.message, 'openReadStream test');
+      });
+  });
+
+  it('should reject on errors in open()', () => {
+    var myXpi = new Xpi('foo/bar', this.fakeZipLib);
+
+    this.openStub.yieldsAsync(new Error('open test'), this.fakeZipFile);
+    return myXpi.getMetaData()
+      .then(() => {
+        assert.fail(null, null, 'Unexpected success');
+      })
+      .catch(err => {
+        assert.include(err.message, 'open test');
+      });
+  });
+});
+
+
+describe('Xpi.getFileAsStream()', function() {
+
+  beforeEach(() => {
+    this.openReadStreamStub = sinon.stub();
+
+    this.fakeZipFile = {
+      openReadStream: this.openReadStreamStub,
+    };
+    this.openStub = sinon.stub();
+    this.fakeZipLib = {
+      open: this.openStub,
+    };
+
+  });
+
+  it('should reject if path does not exist', () => {
+    var myXpi = new Xpi('foo/bar', this.fakeZipLib);
+    myXpi.metadata = {
+      'install.rdf': installRdfEntry,
+      'chrome.manifest': chromeManifestEntry,
+    };
+
+    return myXpi.getFileAsStream('whatever')
+      .then(() => {
+        assert.fail(null, null, 'Unexpected success');
+      })
+      .catch(err => {
+        assert.include(err.message, 'path does not exist');
+      });
+  });
+
+  it('should reject if error in openReadStream', () => {
+    var myXpi = new Xpi('foo/bar', this.fakeZipLib);
+    myXpi.metadata = {
+      'install.rdf': installRdfEntry,
+      'chrome.manifest': chromeManifestEntry,
+    };
+
+    this.openStub.yieldsAsync(null, this.fakeZipFile);
+    this.openReadStreamStub.yields(
+      new Error('getFileAsStream openReadStream test'));
+
+    return myXpi.getFileAsStream('install.rdf')
+      .then(() => {
+        assert.fail(null, null, 'Unexpected success');
+      })
+      .catch(err => {
+        assert.include(err.message, 'getFileAsStream openReadStream test');
+      });
+  });
+
+  it('should resolve with a readable stream', (done) => {
+    var myXpi = new Xpi('foo/bar', this.fakeZipLib);
+    myXpi.metadata = {
+      'install.rdf': installRdfEntry,
+      'chrome.manifest': chromeManifestEntry,
+    };
+
+    this.openStub.yieldsAsync(null, this.fakeZipFile);
+
+    var rstream = new Readable();
+    rstream.push('line one\n');
+    rstream.push('line two');
+    rstream.push(null);
+
+    this.openReadStreamStub.yields(null, rstream);
+
+    myXpi.getFileAsStream('install.rdf')
+      .then((readStream) => {
+        var chunks = '';
+        readStream
+          .on('readable', function () {
+            var chunk;
+            while (null !== (chunk = readStream.read())) {
```

Comment:
That came from the docs example: https://nodejs.org/api/stream.html#stream_readable_read_size